Source/NSURL.m

```diff
@@ -624,70 +639,66 @@ + (id) URLByResolvingAliasFileAtURL: (NSURL*)url
   return nil;
 }
 
-- (id) initFileURLWithPath: (NSString*)aPath
+- (id) initFileURLWithPath: (NSString *)aPath
 {
-  NSFileManager	*mgr = [NSFileManager defaultManager];
-  BOOL		flag = NO;
+  /* isDirectory flag will be overwritten if a directory exists. */
+  self = [self initFileURLWithPath:aPath isDirectory:NO relativeToURL:nil];
```

Comment:
It would be simpler just to return the result of the call. No need to reassign self.

That is:
return [self initFileURLWithPath: aPath isDirectory: NO relativeToURL: nil];